pkg/transformer/kubernetes/kubernetes.go

```diff
@@ -705,37 +708,67 @@ func (k *Kubernetes) Undeploy(komposeObject kobject.KomposeObject, opt kobject.C
 		switch t := v.(type) {
 		case *extensions.Deployment:
 			//delete deployment
-			rpDeployment, err := kubectl.ReaperFor(extensions.Kind("Deployment"), client)
+			label := labels.SelectorFromSet(labels.Set(map[string]string{selector: t.Name}))
```

Comment:
This and next line is same for every object. You can safely move them outside of this `switch` and `for`, than you won't have to repeat same code multiple times 😉